### PR TITLE
Add saving ModuleInfo into the resumption data and updating app hash after subscribe/unsubscribe on a VD

### PR DESCRIPTION
**Pull request by IGapchuk**
Fixes [FORDTCN-7210](https://adc.luxoft.com/jira/browse/FORDTCN-7210)

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Summary
This PR provides the next changes:
- Saving ModuleInfo into the resumption data;
- Updated hash after subscribe/unsubscribe on a VD;

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/get_interior_vehicle_data_request.cc [hunk 1]
```diff
@@ -377,12 +385,16 @@ void GetInteriorVehicleDataRequest::ProccessSubscription(
...
-                        << app->app_id() << " " << module_type << " "
-                        << module_id);
-      extension->UnsubscribeFromInteriorVehicleData(module);
+      if (extension->IsSubscribedToInteriorVehicleData(module)) {
+        LOG4CXX_DEBUG(logger_,
+                      "UnsubscribeFromInteriorVehicleData "
+                          << app->app_id() << " [" << module_type << ":"
+                          << module_id << "]");
+        extension->UnsubscribeFromInteriorVehicleData(module);
+        app->UpdateHash();
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/get_interior_vehicle_data_request.cc:396]: What is the use case of this hash update?

Comment (LuxoftAKutsan) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/get_interior_vehicle_data_request.cc:396]: propose to move app->UpdateHash(); into extension->UnsubscribeFromInteriorVehicleData and SubscribeFromInteriorVehicleData

Comment (IGapchuk) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/get_interior_vehicle_data_request.cc:396]: @LuxoftAKutsan Done in the 9e50bc725ce40f2caa8a2f5ed2e2f7287b02e2d8

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/get_interior_vehicle_data_request.cc [hunk 2]
```diff
@@ -128,14 +128,18 @@ void GetInteriorVehicleDataRequest::ProcessResponseToMobileFromCache(
       auto extension = RCHelpers::GetRCExtension(*app);
       DCHECK(extension);
       extension->SubscribeToInteriorVehicleData(module);
+      app->UpdateHash();
```
Comment (LitvinenkoIra) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/get_interior_vehicle_data_request.cc:131]: @IGapchuk Do we need it here? I can see UpdateHash() here https://github.com/LuxoftSDL/sdl_core/pull/87/files#diff-64546aa83f6f7211621135e409367272R48 in the extension->SubscribeToInteriorVehicleData()

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/get_interior_vehicle_data_request.cc [hunk 3]
```diff
@@ -128,14 +128,18 @@ void GetInteriorVehicleDataRequest::ProcessResponseToMobileFromCache(
...
   }
   SendResponse(
       true, mobile_apis::Result::SUCCESS, nullptr, &response_msg_params);
   if (AppShouldBeUnsubscribed()) {
     auto extension = RCHelpers::GetRCExtension(*app);
     DCHECK(extension);
-    extension->UnsubscribeFromInteriorVehicleData(module);
+    if (extension->IsSubscribedToInteriorVehicleData(module)) {
+      extension->UnsubscribeFromInteriorVehicleData(module);
+      app->UpdateHash();
```
Comment (LitvinenkoIra) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/get_interior_vehicle_data_request.cc:141]: @IGapchuk I believe app->UpdateHash(); is redundant here

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/get_interior_vehicle_data_request.cc (hunk 1 above)
Comment (LitvinenkoIra) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/get_interior_vehicle_data_request.cc:396]: @IGapchuk And please remove UpdateHash() from this func

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/rc_app_extension_test.cc [hunk 4]
```diff
@@ -139,6 +140,48 @@ TEST_F(RcAppExtensionTest,
...
+
+  const auto subscriptions =
+      rc_app_extension_->InteriorVehicleDataSubscriptions();
+
+  auto subscription_so =
+      smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+  rc_app_extension_->SaveResumptionData(subscription_so);
+
+  const uint32_t lenght = subscription_so.length();
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/rc_app_extension_test.cc:168]: also check expected length

Comment (IGapchuk) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/rc_app_extension_test.cc]: Done https://github.com/LuxoftSDL/sdl_core/pull/87/commits/68c0762e57fd9832abdd9c9f665b6ce6f74cb265

**Pull request merged**